agent/job_runner.go

```diff
@@ -503,6 +503,19 @@ func (r *JobRunner) createEnvironment(ctx context.Context) ([]string, error) {
 		env["BUILDKITE_IGNORED_ENV"] = strings.Join(ignoredEnv, ",")
 	}
 
+	// Set BUILDKITE_SECRETS_CONFIG so bootstrap can access secrets configuration
+	if len(r.conf.Job.Step.Secrets) > 0 {
+		if secretsJSON, err := json.Marshal(r.conf.Job.Step.Secrets); err != nil {
+			r.agentLogger.Error("Failed to marshal secrets configuration: %v", err)
+
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			env["BUILDKITE_SECRETS_CONFIG"] = string(secretsJSON)
+		}
```

Comment:
we're double checking the error here and can probably just get rid of the `else`

```suggestion
		secretsJSON, err := json.Marshal(r.conf.Job.Step.Secrets)
		if err != nil {
			r.agentLogger.Error("Failed to marshal secrets configuration: %v", err)
			return nil, err
		}
		
		env["BUILDKITE_SECRETS_CONFIG"] = string(secretsJSON)
```

(one of the benefits of the `if blah, err := func(); err != nil` syntax is that it keeps the scope of both `blah` and `err` solely inside the if block, but they'll be garbage-collected at the end of the outer if block anyway, so there's not much benefit to this scoping)